### PR TITLE
IGNITE-14417 Document performance-statistics-ext module

### PR DESCRIPTION
**Pull request by NSAmelchev**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: docs/_data/toc.yaml [hunk 1]
```diff
@@ -543,6 +543,8 @@
           url: extensions-and-integrations/cassandra/ddl-generator
     - title: PHP PDO
       url: extensions-and-integrations/php-pdo
+    - title: Performance statistics
```
Comment (Nikita-tech-writer) [inline docs/_data/toc.yaml:546]:
```suggestion
    - title: Performance Statistics
```

File: docs/_docs/extensions-and-integrations/performance-statistics.adoc [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+= Performance Statistics Extension
+
+== Overview
+
+Ignite provides link:monitoring-metrics/performance-statistics[a built-in tool for cluster profiling.]
+
+The extension provides utilities to link:#building-the-report[create the report] and
+link:#print-statistics[print statistics] from collected files.
```
Comment (Nikita-tech-writer) [inline docs/_docs/extensions-and-integrations/performance-statistics.adoc:22]:
```suggestion
The extension provides utilities to link:#building-the-report[create the report] and link:#print-statistics[print statistics] from the collected files.
```

File: docs/_docs/extensions-and-integrations/performance-statistics.adoc [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+== Building the Report
+
+Ignite provides a tool to generate the report from performance statistics files.
```
Comment (Nikita-tech-writer) [inline docs/_docs/extensions-and-integrations/performance-statistics.adoc:26]:
```suggestion
Ignite provides a tool to generate the report from the performance statistics files.
```

File: docs/_docs/extensions-and-integrations/performance-statistics.adoc [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+1. Stop collecting statistics and place files from all nodes under an empty directory. For example:
+
+    /path_to_files/
+        ├── node-162c7147-fef8-4ea2-bd25-8653c41fc7fa.prf
+        ├── node-7b8a7c5c-f3b7-46c3-90da-e66103c00001.prf
+        └── node-faedc6c9-3542-4610-ae10-4ff7e0600000.prf
+
+2.  Run the script from the release package of the tool:
+
+        performance-statistics-tool/build-report.sh path_to_files
```
Comment (Nikita-tech-writer) [inline docs/_docs/extensions-and-integrations/performance-statistics.adoc:39]:
```suggestion
[tabs]
--
tab:Unix[]
[source,shell]
----
performance-statistics-tool/build-report.sh path_to_files
----
tab:Windows[]
[source,shell]
----
performance-statistics-tool/build-report.bat path_to_files
----
--
```

File: docs/_docs/extensions-and-integrations/performance-statistics.adoc [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+The performance report is created in the new directory under the performance statistics files
+path: `path_to_files/report_yyyy-MM-dd_HH-mm-ss/`. Open `report_yyyy-MM-dd_HH-mm-ss/index.html` in the browser to see
+the report.
```
Comment (Nikita-tech-writer) [inline docs/_docs/extensions-and-integrations/performance-statistics.adoc:43]:
```suggestion
The performance report is created in the new directory under the performance statistics files: `path_to_files/report_yyyy-MM-dd_HH-mm-ss/`. 
Open `report_yyyy-MM-dd_HH-mm-ss/index.html` in the browser to see the report.
```

File: docs/_docs/extensions-and-integrations/performance-statistics.adoc [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+For more details run the help command:
+
+    performance-statistics-tool/build-report.sh --help
+
+== Print statistics
```
Comment (Nikita-tech-writer) [inline docs/_docs/extensions-and-integrations/performance-statistics.adoc:49]:
```suggestion
== Print Statistics
```

File: docs/_docs/extensions-and-integrations/performance-statistics.adoc [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+        performance-statistics-tool/build-report.sh path_to_files
+
+The performance report is created in the new directory under the performance statistics files
+path: `path_to_files/report_yyyy-MM-dd_HH-mm-ss/`. Open `report_yyyy-MM-dd_HH-mm-ss/index.html` in the browser to see
+the report.
+
+For more details run the help command:
+
+    performance-statistics-tool/build-report.sh --help
```
Comment (Nikita-tech-writer) [inline docs/_docs/extensions-and-integrations/performance-statistics.adoc:47]:
```suggestion
[tabs]
--
tab:Unix[]
[source,shell]
----
performance-statistics-tool/build-report.sh --help
----
tab:Windows[]
[source,shell]
----
performance-statistics-tool/build-report.bat --help
----
--  
```

File: docs/_docs/extensions-and-integrations/performance-statistics.adoc [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+== Print statistics
+
+Ignite provides a tool to print statistics to console or file in JSON format.
```
Comment (Nikita-tech-writer) [inline docs/_docs/extensions-and-integrations/performance-statistics.adoc:51]:
```suggestion
Ignite provides a tool to print statistics to a console or to a file in JSON format.
```

File: docs/_docs/extensions-and-integrations/performance-statistics.adoc [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+Run the script from the release package of the tool to print statistics:
+
+    performance-statistics-tool/print-statistics.sh path_to_files
```
Comment (Nikita-tech-writer) [inline docs/_docs/extensions-and-integrations/performance-statistics.adoc:55]:
```suggestion
[tabs]
--
tab:Unix[]
[source,shell]
----
performance-statistics-tool/print-statistics.sh --path_to_files
----
tab:Windows[]
[source,shell]
----
performance-statistics-tool/print-statistics.bat --path_to_files
----
--  
```

File: docs/_docs/extensions-and-integrations/performance-statistics.adoc [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+where `path_to_files` is a path to performance statistics file or files directory.
```
Comment (Nikita-tech-writer) [inline docs/_docs/extensions-and-integrations/performance-statistics.adoc:78]:
```suggestion
Note that `path_to_files` is a path to the performance statistics file or files directory.
```

File: docs/_docs/extensions-and-integrations/performance-statistics.adoc [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+The script provides the ability to filter operations by operation's type, time, or cache. For more details run the
+help command:
+
+    performance-statistics-tool/print-statistics.sh --help
+
+Example of output:
```
Comment (Nikita-tech-writer) [inline docs/_docs/extensions-and-integrations/performance-statistics.adoc:92]:
```suggestion
See the output example below:
```

File: docs/_docs/extensions-and-integrations/performance-statistics.adoc [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+Run the script from the release package of the tool to print statistics:
+
+    performance-statistics-tool/print-statistics.sh path_to_files
+
+where `path_to_files` is a path to performance statistics file or files directory.
+
+The script provides the ability to filter operations by operation's type, time, or cache. For more details run the
+help command:
+
+    performance-statistics-tool/print-statistics.sh --help
```
Comment (Nikita-tech-writer) [inline docs/_docs/extensions-and-integrations/performance-statistics.adoc]:
```suggestion
[tabs]
--
tab:Unix[]
[source,shell]
----
performance-statistics-tool/print-statistics.sh --help
----
tab:Windows[]
[source,shell]
----
performance-statistics-tool/print-statistics.bat --help
----
-- 
```

**Review by Nikita-tech-writer — COMMENTED** [review]
@NSAmelchev Please see my suggested changes to the PR.
Just in case: You can commit them directly in the PR.

**Comment by NSAmelchev** [comment]
> @NSAmelchev Please see my suggested changes to the PR.
> Just in case: You can commit them directly in the PR.

@Nikita-tech-writer Thank you for the review! I have approved your suggestions.

**Comment by Nikita-tech-writer** [comment]
> > @NSAmelchev Please see my suggested changes to the PR.
> > Just in case: You can commit them directly in the PR.
> 
> @Nikita-tech-writer Thank you for the review! I have approved your suggestions.

Great! I'd like to clarify just one thing:

The script seems work on Windows, doesn't it?

If yes, then it's better to have to separate tabs for Unix and Windows (the ones that I suggested).
Otherwise, the foematting should be as follows:

[source,shell]
----
performance-statistics-tool/print-statistics.sh path_to_files
----

**Comment by NSAmelchev** [comment]
> > > @NSAmelchev Please see my suggested changes to the PR.
> > > Just in case: You can commit them directly in the PR.
> > 
> > 
> > @Nikita-tech-writer Thank you for the review! I have approved your suggestions.
> 
> Great! I'd like to clarify just one thing:
> 
> The script seems work on Windows, doesn't it?
> 
> If yes, then it's better to have two separate tabs for Unix and Windows (the ones that I suggested).
> Otherwise, the foematting should be as follows:
> 
> ## [source,shell]
> ## performance-statistics-tool/print-statistics.sh path_to_files

No, it doesn't work on Windows.

> > > @NSAmelchev Please see my suggested changes to the PR.
> > > Just in case: You can commit them directly in the PR.
> > 
> > 
> > @Nikita-tech-writer Thank you for the review! I have approved your suggestions.
> 
> Great! I'd like to clarify just one thing:
> 
> The script seems to work on Windows, doesn't it?
> 
> If yes, then it's better to have two separate tabs for Unix and Windows (the ones that I suggested).
> Otherwise, the formatting should be as follows:
> 
> ```
> [source,shell]
> ----
> performance-statistics-tool/print-statistics.sh path_to_files
> ----
> ```

@Nikita-tech-writer , No, it doesn't work on Windows. I have fixed tabs.

**Comment by Nikita-tech-writer** [comment]
> @Nikita-tech-writer , No, it doesn't work on Windows. I have fixed tabs.

OK, thanks for your clarification! :)
LGTM!

**Pull request merged**